geoapps/drivers/components/data.py

```diff
@@ -0,0 +1,248 @@
+#  Copyright (c) 2021 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from copy import deepcopy
+from typing import Dict, Tuple
+
+import numpy as np
+from scipy.interpolate import LinearNDInterpolator
+
+from geoapps.utils import filter_xy
+
+from .locations import InversionLocations
+
+
+class InversionData(InversionLocations):
+    """ Retrieve data from workspace and apply transformations. """
+
+    def __init__(self, workspace, params, mesh, topography, window):
+        super().__init__(workspace, params, mesh, window)
+        self.topography = topography
+        self.resolution = None
+        self.offset = None
+        self.radar = None
+        self.ignore_value = None
+        self.ignore_type = None
+        self.detrend_order = None
+        self.detrend_type = None
+        self.components = None
+        self.data = {}
+        self.uncertainties = {}
+        self.normalization = []
+        self.survey = None
+        self._initialize()
+
+    def _initialize(self):
+        """ Extract data from params class. """
+
+        self.ignore_value, self.ignore_type = self.parse_ignore_values()
+        self.components, self.data, self.uncertainties = self.get_data()
+
+        self.locs = super().get_locs(self.params.data_object)
+        self.mask = np.ones(len(self.locs), dtype=bool)
+
+        if self.window is not None:
```

Comment:
I m wondering if this should be moved to the __init__ of InversionLocation since common to topo and data

Reply:
get_locs needs the topo/data specific attribute in self.params so the locs call can't get moved up the hierarchy.  If the locs can't then neither can the mask.

Reply:
Agreed